### ISSUE TITLE
Cannot create fixture in `before` and then use it in multiple tests

### ISSUE THREAD

**Issue by pghalliday** [Polymer/web-component-tester#504]
"web-component-tester": "v6.0.0-prerelease.6"

This took a while to figure out, and then I remembered that while debugging something else and stepping through code I noticed that the fixture `restore` function is automatically added to an `afterEach` handler. This means that code like this will not work (don't worry about the test itself, i'm investigating another problem with `getElementById`)

```javascript
      describe('my-list', function() {
        var myList;

        before(function(done) {
          myList = fixture('myList');
          myList.items = [{
            id: 1,
            value: 'item 1',
          }, {
            id: 2,
            value: 'item 2',
          }, {
            id: 3,
            value: 'item 3',
          }];
          flush(done);
        });

        it('should display the items (querySelector)', function() {
          expect(myList.shadowRoot.querySelector('#item1').textContent).to.eql('item 1');
          expect(myList.shadowRoot.querySelector('#item2').textContent).to.eql('item 2');
          expect(myList.shadowRoot.querySelector('#item3').textContent).to.eql('item 3');
        });

        it('should display the items (getElementById)', function() {
          expect(myList.shadowRoot.getElementById('item1').textContent).to.eql('item 1');
          expect(myList.shadowRoot.getElementById('item2').textContent).to.eql('item 2');
          expect(myList.shadowRoot.getElementById('item3').textContent).to.eql('item 3');
        });
      });
```

When I do this, the myList reference is reset or something before the second test runs - most surprising! I can workaround it by making my `before` a `beforeEach` but it's not pretty ;)

**Comment by rictic** [comment]
I believe part of fixture's raison d'etre is to automatically clean itself up after each test runs. So this is working as intended, and calls to `fixture` shouldn't happen in `before`, only `beforeEach`.

**Comment by pghalliday** [comment]
I'll be honest, I have no idea why you think the raison d'etre of a fixture is to clean up after itself. From my point of view this behavior seems awfully opinionated to the extent that it prevents common styles of writing unit tests. Ie run a scenario, write multiple tests to verify the behavior - it can be verbose but you don't always want tons of assertions in a single test.

**Comment by rictic** [comment]
The danger in that style of testing is that unless you are very disciplined and careful in not doing any mutation in your tests you'll end up with tests influencing each other when they should be standalone. This is why test-fixture is designed to clean itself up after each test, in order to provide test isolation. It is a bit magical, and I'm sorry that that bit you here, there's a tension between sugary magic and verbose clarity.

Here's where the feature is documented, as we use it in wct: https://github.com/PolymerElements/test-fixture#even-simpler-usage-in-mocha

**Comment by pghalliday** [comment]
Thanks for the link :)

I was starting to get confused again as I have other tests that use this pattern that pass but then I realised that I was not mutating the fixture after creating it and as a result the `restore` is harmless. The only reason I was mutating the fixture in the above snippet is because I hadn't figured out how to set an attribute to an array before creating it (I suspect a JSON string might work but didn't try it as it would be pretty ugly/cumbersome). Looks like I may be able to use a data bound template to achieve this.

**Comment by pghalliday** [comment]
Data binding didn't work either as it also gets reset I suppose. Just on the off chance I tried a JSON string but that didn't work either. So the only way to get this to work with an array attribute and `dom-repeat` is to use `beforeEach` as far as I can tell.

I think the reason is that when the fixture is restored it is done synchronously without a `flush`, this means that the next test runs before my `dom-repeat` renders. I noticed that when I had a static element bound to data it rendered correctly before the next test runs.

Lastly, and this may be more important. I couldn't get data bound fixtures to work at all with the Polymer 2.0 RC (test-fixture v3.0.0-rc.1) - is this a known issue, i couldn't find it?
